File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <map>
#include <set>
using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    
    map<string, int> report_cnt;
    map<string, set<string>> report_list;

    for(auto s : report){
        int blank = s.find(' ');
        // 신고한 사람
        string from = s.substr(0, blank);
        // 신고받은 사람
        string to = s.substr(blank+1, s.length());

        // set 요소 find ... 중복 거르기
        if(report_list[from].find(to) == report_list[from].end()){
            report_cnt[to]++;
            report_list[from].insert(to);
        }
    }

    for(auto s : id_list){
        int cnt = 0;

        // id_list마다 set에 신고한 id가 저장되어 있으므로
        for(string str : report_list[s]){
            if(report_cnt[str] >= k)
                cnt++;
        }

        answer.push_back(cnt);
    }
    
    return answer;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    var answer = [];
    
    // id_list 크기만큼 0 삽입
    id_list.forEach(()=>{
        answer.push(0);
    });
    
    // id_list 크기만큼 report_list에 신고받은 사람, 배열 삽입
    let report_list = {};
    id_list.map((s)=>{
    	return report_list[s] = [];
    });
    // 
    report.map((user)=>{
        // 0 신고한 사람, 1 신고받은 사람
    	let arr = user.split(" ");
		if(!report_list[arr[1]].includes(arr[0]))
        	report_list[arr[1]].push(arr[0]);
    });
    
    for(let list in report_list){
        // 신고한 사람이 k보다 많으면
    	if(report_list[list].length >= k){
        	report_list[list].map((arr)=>{
            	answer[id_list.indexOf(arr)]++;
            });
        }
    }
    
    return answer;
}